src/orquestra/sdk/_base/cli/_dorq/_login/_login.py

```diff
@@ -37,26 +42,67 @@ def __init__(
         self._config_repo: _repos.ConfigRepo = config_repo
         self._runtime_repo: _repos.RuntimeRepo = runtime_repo
         self._login_server: LoginServer = login_server
+        self._config_resolver: _arg_resolvers.ConfigResolver = config_resolver
 
-    def on_cmd_call(self, url: str, token: t.Optional[str], ce: bool):
+    def on_cmd_call(
+        self,
+        config: t.Optional[str],
+        url: t.Optional[str],
+        token: t.Optional[str],
+        ce: bool,
+    ):
+        """
+        Call the login command action, catching any exceptions that arise.
+        """
         try:
-            self._on_cmd_call_with_exceptions(url, token, ce)
+            self._on_cmd_call_with_exceptions(config, url, token, ce)
         except Exception as e:
             self._exception_presenter.show_error(e)
 
     def _on_cmd_call_with_exceptions(
         self,
-        url: str,
+        config: t.Optional[str],
+        url: t.Optional[str],
         token: t.Optional[str],
         ce: bool,
     ):
         """
         Implementation of the command action. Doesn't catch exceptions.
         """
+        assert bool(config) ^ bool(url), (
```

Comment:
Suggestion: we have _exception_presenter to show errors like this to the user. Consider just throwing it here and using presenter for anything UI related

Reply:
(added as suggestion non-blocking, as this indeed should never happen with cloup restriction btw)

Reply:
Good point, I'm keeping most of the text here as it makes it clear to us exactly what's gone wrong as and when somebody reports it, but removing the 'please report this as a bug' line since the presenter adds that anyway.